ui/dashboards/src/components/PanelDrawer/PanelDrawer.tsx

```diff
@@ -14,43 +14,55 @@
 import { useState } from 'react';
 import { Stack, Box, Button, Typography } from '@mui/material';
 import { Drawer } from '@perses-dev/components';
-import { useDashboardApp } from '../../context';
-import { usePanelDrawerModel } from './panel-editor-model';
+import { PanelDefinition } from '@perses-dev/core';
+import { usePanels } from '../../context';
 import { PanelEditorForm, panelEditorFormId, PanelEditorFormProps } from './PanelEditorForm';
 
 /**
  * The Add/Edit panel drawer for editing a panel's options.
  */
 export const PanelDrawer = () => {
-  const { closePanelDrawer } = useDashboardApp();
-  const model = usePanelDrawerModel();
+  const { panelEditor } = usePanels();
 
   // When the user clicks close, start closing but don't call the store yet to keep values stable during animtation
   const [isClosing, setIsClosing] = useState(false);
   const handleClose = () => setIsClosing(true);
 
   // Don't call closeDrawer on the store until the Drawer has completely transitioned out
   const handleExited = () => {
-    closePanelDrawer();
+    panelEditor?.close();
```

Comment:
I wish! I already tried to do that for a lot of the logic by moving the form/header into something conditionally rendered below. Unfortunately because this is a drawer that's always rendered whether it's open or not, there's some limits to what we can do without just conditionally rendering this entire component anywhere it's used. The animation transition for it appearing/disappearing also complicates things. 😢

Reply:
Feels like an issue for a different time, but I wonder if there's a way to get Drawer to _not_ always render

Reply:
Would definitely be cool if there's a pattern for solving the transition issue more generally because it comes up a lot. The issue is really on transition out, where you want the UI to stay "stable" (which typically means you want the props being passed to it to largely stay "stable") while it's animating out.